packages/react/src/components/Authenticator/Router/Router.tsx

```diff
@@ -79,9 +69,22 @@ export function Router({
   hideSignUp,
   variation,
 }: RouterProps) {
-  const { route } = useAuthenticator(({ route }) => [route]);
+  const { route, signOut, user } = useAuthenticator(
+    ({ route, signOut, user }) => [route, signOut, user]
+  );
   const RouterChildren = useRouterChildren(children);
 
+  // `Authenticator` might not have user defined `children` for non SPA use cases.
+  if (route === 'authenticated' || route === 'signOut') {
+    if (!children) {
+      return null;
+    }
+
+    return typeof children === 'function'
+      ? children({ signOut, user }) // children is a render prop
+      : children;
+  }
```

Comment:
I moved this code out of the React.memo since it was causing a rerender whenever the input changed.

Reply:
And that seemed to be causing the input to lose focus

Reply:
Curious, why is `children` changing on input events?

Reply:
I know the children have an input, that is tied to an `onChange` event, that updates `useState`. I'm not sure what's causing the `useMemo` to fire though and cause the issue.

Reply:
Think we need to root cause the re-render before moving the code out of the `useMemo`. Not saying that this  is the wrong approach, but we should know for certain that this is the actual cause. 

If we are going to move forward with this approach we will want to remove the duplicate logic in the `useRouterChildren` hook, or consider removing the hook altogether

Reply:
I can create a utility function that returns back `isUnauthenticatedRoute`. 

OK, I tested some more. From my testing `useMemo` is rerendering anytime the `children` change. That's causing the input to lose focus on every key press. It only occurs when the children have an input that is using a `useState` hook.

Reply:
#5577559816d0be32fbbc96e70f0de8c8b5eece3b

Reply:
Still a bit confused. The `useMemo` **should** update when the value of `children` changes, but not sure why `children` would change due to input events